src/Codec/Xlsx/StyleSheet.hs

```diff
@@ -0,0 +1,1123 @@
+    -- 180. The first letter of the text is considered the center-point of the
+    -- arc.
+  , _alignmentTextRotation :: Maybe Int
+
+    -- | Vertical alignment in cells.
+  , _alignmentVertical :: Maybe CellVerticalAlignment
+
+    -- | A boolean value indicating if the text in a cell should be line-wrapped
+    -- within the cell.
+  , _alignmentWrapText :: Maybe Bool
+  }
+  deriving (Show, Eq, Ord)
+
+-- | Expresses a single set of cell border formats (left, right, top, bottom,
+-- diagonal). Color is optional. When missing, 'automatic' is implied.
+--
+-- See 18.8.4 "border (Border)" (p. 1949)
+data Border = Border {
+    -- | A boolean value indicating if the cell's diagonal border includes a
+    -- diagonal line, starting at the top left corner of the cell and moving
+    -- down to the bottom right corner of the cell.
+    _borderDiagonalDown :: Maybe Bool
+
+    -- | A boolean value indicating if the cell's diagonal border includes a
+    -- diagonal line, starting at the bottom left corner of the cell and moving
+    -- up to the top right corner of the cell.
+  , _borderDiagonalUp :: Maybe Bool
+
+    -- | A boolean value indicating if left, right, top, and bottom borders
+    -- should be applied only to outside borders of a cell range.
+  , _borderOutline :: Maybe Bool
+
+    -- | Bottom border
+  , _borderBottom :: Maybe BorderStyle
+
+    -- | Diagonal
+  , _borderDiagonal :: Maybe BorderStyle
+
+    -- | Trailing edge border
+    --
+    -- See also 'borderRight'
+  , _borderEnd :: Maybe BorderStyle
+
+    -- | Horizontal inner borders
+  , _borderHorizontal :: Maybe BorderStyle
+
+    -- | Leading edge border
+    --
+    -- See also 'borderLeft'
+  , _borderStart :: Maybe BorderStyle
+
+    -- | Top border
+  , _borderTop :: Maybe BorderStyle
+
+    -- | Vertical inner border
+  , _borderVertical :: Maybe BorderStyle
+
+    -- | Left border
+    --
+    -- NOTE: The spec does not formally support a 'left' border, but the
+    -- examples do mention 'left'. See also 'borderStart'.
+  , _borderLeft :: Maybe BorderStyle
+
+    -- | Right border
+    --
+    -- NOTE: The spec does not formally support a 'right' border, but the
+    -- examples do mention 'right'. See also 'borderEnd'.
+  , _borderRight :: Maybe BorderStyle
+  }
+  deriving (Show, Eq, Ord)
+
+-- | Border style
+data BorderStyle = BorderStyle {
+    _borderStyleColor :: Maybe Color
+  , _borderStyleLine  :: Maybe LineStyle
+  }
+  deriving (Show, Eq, Ord)
+
+-- | One of the colors associated with the data bar or color scale.
+--
+-- The 'indexed' attribute (used for backwards compatibility only) is not
+-- modelled here.
+--
+-- See 18.3.1.15 "color (Data Bar Color)" (p. 1780)
+data Color = Color {
+    -- | A boolean value indicating the color is automatic and system color
+    -- dependent.
+    _colorAutomatic :: Maybe Bool
+
+    -- | Standard Alpha Red Green Blue color value (ARGB).
```

Comment:
That's cool - spec doesn't mention alpha channel at all while states that it should be 8 hex digits and gives examples with 8 digits (but with FF for alpha) and 1 example with `<xs:bgColor type="rgb" val="4278252287"/>` (not FF alpha but in decimal form)